packages/xai/src/convert-to-xai-chat-messages.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  LanguageModelV1Prompt,
+  UnsupportedFunctionalityError,
+} from '@ai-sdk/provider';
+import { convertUint8ArrayToBase64 } from '@ai-sdk/provider-utils';
+import { XaiChatPrompt } from './xai-api-types';
+
+export function convertToXaiChatMessages(
+  prompt: LanguageModelV1Prompt,
+): XaiChatPrompt {
+  const messages: XaiChatPrompt = [];
+
+  for (const { role, content } of prompt) {
+    switch (role) {
+      case 'system': {
+        messages.push({ role: 'system', content });
+        break;
+      }
+
+      case 'user': {
+        if (content.length === 1 && content[0].type === 'text') {
+          messages.push({ role: 'user', content: content[0].text });
+          break;
+        }
+
+        messages.push({
+          role: 'user',
+          content: content.map(part => {
+            switch (part.type) {
+              case 'text': {
+                return { type: 'text', text: part.text };
+              }
+              case 'image': {
+                return {
+                  type: 'image_url',
+                  image_url: {
+                    url:
+                      part.image instanceof URL
+                        ? part.image.toString()
+                        : `data:${
+                            part.mimeType ?? 'image/jpeg'
+                          };base64,${convertUint8ArrayToBase64(part.image)}`,
+                  },
```

Comment:
Does grok support images?

Reply:
It doesn't seem to currently:
- nothing in API docs
- attempting with the `generate-text` example produces: `'{"code":"Client specified an invalid argument","error":"The model does not support content elements of type image."}'`

Plan:
- edit docs to turn the check to a cross for image support
- change conversion code to behave like other providers that don't support images (presuming we throw but will validate/emulate)
- @lgrammel - let them know we don't see support for images on the shared Slack channel (or let me know if I should do it, best to have one person talking w/ them though IMO)